### PR TITLE
Vttest

### LINKED ISSUE #120: Make vttest test 1 pass
Test 1 of the `vttest` tool needs to pass.

This is usually installed by default on most systems.

See #88 for more information

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
# Pull Request Template

## Description

Correctly handles most of the screens of vttest test 1

Fixes #120 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Run `vttest`. Run test 1 (cursor movements)

**Test Configuration**:
* OS: Ubuntu
* OS version: 18.04
* Go version: 1.11

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by MaxRis** [comment]
@rrrooommmaaa looks like some test is failing on Travis https://travis-ci.org/liamg/aminal/jobs/481534141#L1051

**Comment by MaxRis** [comment]
@rrrooommmaaa tests output looks great! But the terminal Window doesn't resize to "80x40" for test 1.1 and to "132x40" for test 1.2 (similar issue for 1.3 and 1.4 tests). Interesting that both standard Terminal on Ubuntu and Terminal.app on MacOS are doing Window resizing, but iTerm2.app fails test case because of that as well. xTerm resizes Windows correctly as well.

**Comment by rrrooommmaaa** [comment]
@MaxRis I have done window resize, now your check doesn't pass. Can you have a look, please?

File: buffer/buffer.go [hunk 1]
```diff
@@ -747,21 +786,34 @@ func (buffer *Buffer) NewLine() {
...
 
+func (buffer *Buffer) SetNewLineMode() {
+	buffer.lineFeedMode = false
+}
+
+func (buffer *Buffer) SetLineFeedMode() {
+	buffer.lineFeedMode = true
+}
+
+func (buffer *Buffer) NewLineMode() bool {
```
Comment (MaxRis) [inline buffer/buffer.go]: will it fit better to name the method `IsNewLineMode()`?

File: terminal/output.go [hunk 2]
```diff
@@ -8,66 +8,85 @@ import (
...
+	0x09: tabHandler,
+	0x0a: newLineHandler,
+	0x0b: newLineHandler,
+	0x0c: newLineHandler,
+	0x0d: carriageReturnHandler,
+	0x0e: shiftOutHandler,
+	0x0f: shiftInHandler,
+}
+
 var escapeSequenceMap = map[rune]escapeSequenceHandler{
```
Comment (MaxRis) [inline terminal/output.go]: map for 1 element, probably, is not required anymore

File: buffer/buffer.go (hunk 1 above)
Comment (rrrooommmaaa) [reply buffer/buffer.go]: This was what I originally wrote but it won't fit with AutoWrap() identifier, will it?

Comment (rrrooommmaaa) [reply buffer/buffer.go]: @MaxRis  oh, I haven't committed that yet.
I got the idea. The private member is autoWrap, the public is IsAutoWrap(), The private is newLineMode, the public is IsnewLineMode(), right?

Comment (MaxRis) [reply buffer/buffer.go]: This should work! Thanks!

**Comment by rrrooommmaaa** [comment]
@MaxRis The test doesn't pass if run subsequently 2 times (because of Line Mode turned on, etc.) So I suggest you to add more screens to the test.

**Comment by MaxRis** [comment]
@rrrooommmaaa do you mean that it makes sense to circle running of test 1 for multiple times? It seems the same screenshots can be used for validation

**Comment by rrrooommmaaa** [comment]
@MaxRis Yes, it makes sense to run test 1 two times consequently.

**Pull request merged**